### PR TITLE
[Issue #8] MedicalAppointment form

### PR DESCRIPTION
**Pull request by piotrgos45**
created activity_medical_appointment_form.xml
created MedicalAppointmentFormActivity.kt
removed appointmentViewModel and appointmentViewModelFactory from MainActivity.kt

### CODE REVIEW

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                    this,
+                    "Please enter a date",
+                    Toast.LENGTH_SHORT
+                ).show()
+            else if (binding.editTextTime.text.isNullOrEmpty())
+                Toast.makeText(
+                    this,
+                    "Please enter a time",
+                    Toast.LENGTH_SHORT
+                ).show()
```
Comment (kozych) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:96]: This is fine but also try to use TextWatcher to validate this fields
https://betterprogramming.pub/textinputlayout-form-validation-using-data-binding-in-android-86aea8645a11

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            else {
+                val title = binding.editTextTitle.text.toString()
+                val day = calendar.get(Calendar.DAY_OF_MONTH)
+                val month = calendar.get(Calendar.MONTH)
+                val year = calendar.get(Calendar.YEAR)
+                val facility = binding.editTextMedicalFacility.toString()
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:102]: Check out what do you pass here. You should get the .text.toString() from the editText instead.

Comment (piotrgos45) [reply app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt]: done

**Review by kozych — APPROVED** [review]

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            binding.textInputLayoutTime.isErrorEnabled = false
+        }
+
+        return true
+    }
+
+    private fun isValidate(): Boolean =
+        validateTitle() && validateTextMedicalFacility() && validateDate() && validateTime()
+
+    private fun setupListeners() {
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:109]: this is not called anywhere

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+        val appointmentViewModelFactory = AppointmentViewModelFactory(
+            (application as PatientApplication).appointmentRepository
+        )
+        appointmentViewModel = ViewModelProvider(
+            this, appointmentViewModelFactory
+        )[AppointmentViewModel::class.java]
+
+        binding.editTextDate.setOnClickListener {
+            DatePickerDialog(
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:138]: There's a problem when the user have 24 hour format set, the hour is saved in the database as 12h format.

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            }
+            else {
+                TimePickerDialog(
+                    this,
+                    timeSetListener,
+                    calendar.get(Calendar.HOUR_OF_DAY),
+                    calendar.get(Calendar.MINUTE),
+                    false
+                ).show()
+            }
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:173]: This should work 👍🏼

**Comment by michalmobi** [comment]
@AlicjaKowalska please review

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    private val calendar = Calendar.getInstance()
+    private lateinit var dateSetListener: DatePickerDialog.OnDateSetListener
+    private lateinit var timeSetListener: TimePickerDialog.OnTimeSetListener
+
+    lateinit var appointmentViewModel: AppointmentViewModel
+    lateinit var binding: ActivityMedicalAppointmentFormBinding
+
+    inner class TextFieldValidation(private val view: View) : TextWatcher {
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            TODO("Not yet implemented")
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:33]: Please implement not implemented functions

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        )
+        binding.editTextTime.addTextChangedListener(
+            TextFieldValidation(binding.editTextTime)
+        )
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupListeners()
```
Comment (AlicjaKowalska) [inline app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:128]: this should be called after binding the view

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt (hunk 6 above)
Comment (piotrgos45) [reply app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt:33]: done

File: app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt (hunk 7 above)
Comment (piotrgos45) [reply app/src/main/java/com/example/patientassistant/View/MedicalAppointmentFormActivity.kt]: done

**Review by AlicjaKowalska — APPROVED** [review]

**Review by michalmobi — APPROVED** [review]

**Pull request merged**